include/circt/Dialect/FIRRTL/FIRRTLTypes.td

```diff
@@ -46,6 +46,14 @@ def UInt1Type : DialectType<FIRRTLDialect,
    "UInt<1> or UInt", "::circt::firrtl::UIntType">,
    BuildableType<"UIntType::get($_builder.getContext(), 1)">;
 
+def UInt01Type : DialectType<FIRRTLDialect,
+  CPred<"$_self.isa<UIntType>() && "
+        "($_self.cast<UIntType>().getWidth() == 0 ||"
+        "$_self.cast<UIntType>().getWidth() == 1 ||"
+        " $_self.cast<UIntType>().getWidth() == None)">,
+   "UInt<0> UInt<1> or UInt", "::circt::firrtl::UIntType">,
+   BuildableType<"UIntType::get($_builder.getContext(), 1)">;
+
```

Comment:
Nice to move this into the tables!

Reply:
Thanks! Yeah, this is pretty clean. I'm not super happy that there's `UIntType`, `UInt1Type`, and `UInt01Type`. This explicit monomorphisation in tablegen isn't awesome, however. 🤷‍♀️ This could probably be cleaned up with a template here.

Reply:
What do you mean by monomorphisation?  If the large expression is bugging you, you could move this out to a helper function in FIRRTLTypes.h/cpp and implement the predicate by calling the function.  I think it is ok either way

Reply:
It smells that I'm having to enumerate different flavors of `UIntType` with different allowable widths. It just seems inelegant. I'm totally fine with the large expression, just slightly concerned with the blowup of extra types.